articles/connections/apple-siwa/add-siwa-to-native-app.md

```diff
@@ -0,0 +1,75 @@
+---
+title: Add Sign In with Apple to Native iOS Apps
+description: Learn how to add native login functionality to your native app with Apple. 
+topics:
+  - authentication
+  - connections
+  - social
+  - apple
+contentType: how-to
+useCase:
+  - add-login
+  - customize-connections
+  - add-idp
+---
+# Add Sign In with Apple to Native iOS Apps
+
+You can add functionality to your native iOS application that allows your users to use Sign In with Apple to authenticate. In addition, you can also try the Auth0 [iOS Swift - Sign In with Apple Quickstart](/quickstart/native/ios-swift-siwa).
+
+## How it works
+
+For a native app, the Sign in with Apple login flow works as follows:
+
+![Sign In with Apple Authentication Flow](/media/articles/connections/social/apple/apple-siwa-authn-flow.png)
+
+* User authenticates via Apple's SDK on their iPhone or iPad. They receive an authorization code. The user does not have to leave the app and use a browser to log in.
+* The application calls Auth0's `/oauth/token` endpoint with the following parameters:
+    - `subject_token`: the authorization code they received above
+    - `subject_token_type`: `http://auth0.com/oauth/token-type/apple-authz-code`
+    - `grant_type`: `urn:ietf:params:oauth:grant-type:token-exchange`
+    - `client_id`: their Auth0 Client ID
+    - `audience` and `scope` as needed (optional)
+
+    Auth0 exchanges the `subject_token` (authorization code) with Apple for an ID token, access token, and refresh token from Apple.
+* Auth0 saves the user profile, executes rules and authorization, then issues Auth0 access tokens (refresh tokens and ID tokens) as requested. These tokens are used to protect your APIs and users managed by Auth0.
+
+## Prerequisites
```

Comment:
So - just curious.  We mention custom domain/email purposes in connection settings, but don't list that as a prerequisite here.  Email relay configs are the same (w/ domain validation) for web app configs as they are for native.  Not sure why it's one place and not the other.  Could we look to make the "Prerequisites" sections a little more consistent with one another?

Reply:
Agreed - I'll make it more consistent.